common/thorhelper/thorsoapcall.cpp

```diff
@@ -1362,30 +1377,43 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
         else
             request.clear().append(master->service).append(" ").append(url.path).append(" HTTP/1.1\r\n");
 
-        if (url.userPasswordPair.length() > 0)
+        const char *httpheaders = master->httpHeaders.get();
+        if (httpheaders && *httpheaders)
         {
-            StringBuffer authToken;
-            JBASE64_Encode(url.userPasswordPair.str(), url.userPasswordPair.length(), authToken);
-            request.append("Authorization: Basic ").append(authToken).append("\r\n");
+            if (soapTraceLevel > 6 || master->logXML)
+                master->logctx.CTXLOG("%s: Adding HTTP Headers(%s)",  master->wscCallTypeText(), httpheaders);
+            request.append(httpheaders);
         }
-        else if (master->authToken.length() > 0)
+
+        if (!httpHeaderBlockContainsHeader(httpheaders, "Authorization"))
         {
-            request.append("Authorization: Basic ").append(master->authToken).append("\r\n");
+            if (url.userPasswordPair.length() > 0)
+            {
+                StringBuffer authToken;
+                JBASE64_Encode(url.userPasswordPair.str(), url.userPasswordPair.length(), authToken);
+                request.append("Authorization: Basic ").append(authToken).append("\r\n");
+            }
+            else if (master->authToken.length() > 0)
+            {
+                request.append("Authorization: Basic ").append(master->authToken).append("\r\n");
+            }
         }
+
         if (master->wscType == STsoap)
         {
             if (master->soapaction.get())
                 request.append("SOAPAction: ").append(master->soapaction.get()).append("\r\n");
-
-            if (master->httpHeaderName.get() && master->httpHeaderValue.get())
+            if (!master->httpHeaders.length() && master->httpHeaderName.get() && master->httpHeaderValue.get())
```

Comment:
trivial: master->httpHeaders.isEmpty() is more efficient.  Not significant in this situation though.